### ISSUE TITLE
I cannot achieve success in classification training 

### ISSUE THREAD

**Issue by ooguzzzz** [ultralytics/ultralytics#8077]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I want to do classification training on rectangular images.
However, in the batch images created during training, I see that the images are turned into squares and the centre is taken exactly.
The training results are unsuccessful, are these images taken as square for training, or do they appear like this in batch images.
If it is square, what should I do to prevent this?
I turned off all augmentations (I think) but nothing changed.
![0](https://github.com/ultralytics/ultralytics/assets/101019436/702a1350-7f33-4800-b3d9-733d5a2e6c5e)
![0](https://github.com/ultralytics/ultralytics/assets/101019436/eb3afdfc-36a8-41c5-a7b7-db57ff18775b)
![0](https://github.com/ultralytics/ultralytics/assets/101019436/d77d2b06-3d92-440a-ad6b-b8a93cf07aa6)
![0](https://github.com/ultralytics/ultralytics/assets/101019436/422c54ca-f12d-4f1d-a6a0-8ea2032ecb64)
![3](https://github.com/ultralytics/ultralytics/assets/101019436/630b6641-9386-432a-950e-51b36bdf18fd)

![2](https://github.com/ultralytics/ultralytics/assets/101019436/d6dc4f00-9c51-40b5-80ea-d164ee4207e4)

**Comment by glenn-jocher** [comment]
@ooguzzzz it looks like your images are being resized to squares during training, which is the default behavior for many object detection models, including YOLOv8. To train on rectangular images without distortion, you can modify the `--imgsz` argument to match your desired width and height, for example `--imgsz 640 480` for a 640x480 resolution. Additionally, ensure that the `--rect` training flag is set to `True` to enable rectangular training. If you've already done this and still face issues, could you please share your training command and dataset configuration file? This will help us understand the problem better. 🛠️

**Comment by ooguzzzz** [comment]
I run the code in CLI. When I wrote --imgsz 640 480, it gave an error. Then I ran it by making imgsz=[400,70]. Although the Rect setting is True and I set the required size for imgsz, it sets it to 416, saying that the imgsz value must be int for train during training. Is there another way to do this? Or does it make sense to add black padding to the bottom of the images and make them square? If I do the training this way, do I need to add black padding in the test?
